addons/binding/org.openhab.binding.silvercrestwifisocket/src/main/java/org/openhab/binding/silvercrestwifisocket/internal/utils/WifiSocketPacketConverter.java

```diff
@@ -207,9 +212,21 @@ private SilvercrestWifiSocketResponse decryptResponsePacket(final String hexPack
                     + "]  \nDecryptedPacket:[" + decryptedData + "]");
         }
 
-        logger.trace("Decrypt success. Packet is from socket with mac address [{}] and type is [{}]", macAddress,
-                responseType);
-        return new SilvercrestWifiSocketResponse(macAddress, responseType);
+        SilvercrestWifiSocketVendor vendor = null;
+        for (SilvercrestWifiSocketVendor v : SilvercrestWifiSocketVendor.values()) {
+            if (v.getCompanyCode().equals(decryptedData.substring(6, 8))) {
+                vendor = v;
+                break;
+            }
+        }
+        if (vendor == null) {
+            throw new PacketIntegrityErrorException("Could not extract vendor from the decrypted packet. \nPacket:["
+                    + hexPacket + "]  \nDecryptedPacket:[" + decryptedData + "]");
+        }
+
+        logger.trace("Decrypt success. Packet is from socket with mac address [{}] and type is [{}] and vendor is [{}]",
```

Comment:
You don't have to call toString it will be called automatically, but only when the logging-level actually applies.

Reply:
Will remove the toString